rfd/0089-merge-webapps.md

```diff
@@ -0,0 +1,185 @@
+---
+authors: Jeff Pihach (jeff.pihach@goteleport.com)
+state: draft
+---
+
+# RFD 89 - Merge Webapps into Teleport
+
+## What
+
+- Merge the [webapps](https://github.com/gravitational/webapps) repo into the
+  [teleport](https://github.com/gravitational/teleport) repo.
+- Merge the [webapps.e](https://github.com/gravitational/webapps.e) repo into
+  the [teleport.e](https://github.com/gravitational/teleport.e) repo.
+
+## Why
+
+The version of the web UI and the Teleport are tightly coupled but the
+repositories were created separate to support both Gravity and Teleport. This
+introduced complexities when integrating the two projects.
+
+We have to separately clone and build the web UI and commit the webasset
+changes into the teleport repo whenever we want to represent an update. We also
+have to maintain multiple build systems and ops procedures for the web UI.
+
+Now that Gravity is no longer a product we develop, moving the web UI
+repositories into their respective Teleport repo will save us time and
+confusion.
+
+## Success Criteria
+
+Both [Teleport](https://github.com/gravitational/teleport) and
+[Teleport.e](https://github.com/gravitational/teleport.e) repos contain their
+respective UI counterparts and the build systems correctly build all projects.
+
+## Developer Experience
+
+### Day to day
+
+The Day to day development experience shouldn't change. The webapps project
+will continue to be self-contained and only require access to the nodejs and
+yarn binaries. The locations of the enterprise assets will have to be changed
+but that should remain transparent to the day to day activities.
+
+To work on the enterprise version of the application the developer will have to
+check out the enterprise version of Teleport which will bring with it the
+appropriate webapp assets.
+
+In the event that multiple developers are working on the same feature across
+the back and frontend work may need to be done on the same branch or the
+developer may want to clone Teleport into another folder to build and update
+the backend binaries separate from the frontend.
+
+### Backports
+
+To avoid having different proceses and build systems depending on the version
+of teleport, all branch versions (v9, v10, v11) will have their respective
+versions of the webapps repository.
```

Comment:
Does this mean that the merge of webapps into teleport will be backported to v8, v10 and v11?

Reply:
Yes that's correct, well v9-11 :)